extensions/amp-viewer-gpay-inline/0.1/amp-viewer-gpay-inline.js

```diff
@@ -13,35 +13,64 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+import {
+  AmpPaymentGoogleInlineService,
+  AmpPaymentGoogleIntegration,
+} from '../../../src/service/payments/amp-payment-google';
+import {CSS} from '../../../build/amp-viewer-gpay-inline-0.1.css';
+import {getServiceForDoc} from '../../../src/service';
+import {isLayoutSizeDefined} from '../../../src/layout';
 
-import {Layout} from '../../../src/layout';
+/** @const {string} */
+const TAG = 'amp-viewer-gpay-inline';
 
+/** @implements {../../../src/async-input.AsyncInput} */
 export class AmpViewerGpayInline extends AMP.BaseElement {
   /** @param {!AmpElement} element */
   constructor(element) {
     super(element);
-
-    /** @private {string} */
-    this.myText_ = 'hello world';
-
-    /** @private {?Element} */
-    this.container_ = null;
   }
 
   /** @override */
   buildCallback() {
-    this.container_ = this.element.ownerDocument.createElement('div');
-    this.container_.textContent = this.myText_;
-    this.element.appendChild(this.container_);
-    this.applyFillContent(this.container_, /* replacedContent */ true);
+    /**
+     * @private {!AmpPaymentGoogleIntegration}
+     */
+    this.paymentsIntegration_ = getServiceForDoc(
+      this.win.document,
+      'amp-payment-google-integration'
+    );
+    this.paymentsIntegration_.startInlinePayment(this.element);
+  }
+
+  /** @override */
+  layoutCallback() {
+    return this.paymentsIntegration_.whenInlineWidgetReady();
   }
 
   /** @override */
   isLayoutSupported(layout) {
-    return layout == Layout.RESPONSIVE;
+    return isLayoutSizeDefined(layout);
```

Comment:
Similarly, make this consistent with the layouts supported in the validator file at: 
https://github.com/ampproject/amphtml/blob/05c41cd2d15b871fdb0e0882720c5a37bafdd3e5/extensions/amp-viewer-gpay-inline/validator-amp-viewer-gpay-inline.protoascii#L33-L35

Reply:
got it. to clarify, for now, the only allowed layout will be responsive, but once the followup validator changes are made, the layouts will need to be updated to the isLayoutSizeDefined?

Reply:
Yup, that is correct.